### PR TITLE
Force Check Lookup In HttpTrap Callback

### PR DESCRIPTION
**Pull request by pamaddox**
Every time we get into the HttpTrap callback, force looking up the check
by the uuid. This will prevent us from trying to use a stale check
pointer.

### CODE REVIEW

File: src/modules/httptrap.c [hunk 1]
```diff
@@ -457,6 +458,12 @@ rest_get_json_upload(mtev_http_rest_closure_t *restc,
 
   content_length = mtev_http_request_content_length(req);
   rxc = restc->call_closure;
+  rxc->check = noit_poller_lookup(rxc->check_id);
+  if (!rxc->check) {
+    *complete = 1;
```
Comment (aidancully) [inline src/modules/httptrap.c]: should we report an error, here?

Comment (aidancully) [inline src/modules/httptrap.c]: should we report an error, here?

Comment (pamaddox) [inline src/modules/httptrap.c]: This isn't necessarily an error.... it likely means that a check was deleted while in the middle of processing and that the rest of the data should be discarded. I think we could potentially spam too much stuff in the error log here if people constantly create and delete a bunch of httptrap checks.

Comment (aidancully) [inline src/modules/httptrap.c]: I meant to clients... I guess an ```
rxc->error = strdup("check invalidated during processing");
``` or something like that?

Comment (aidancully) [inline src/modules/httptrap.c]: oic, we report "internal error" in that case, nvm.

**Comment by aidancully** [comment]
LGTM

**Pull request merged**